2016/proposals/stolliker-ryan.md

```diff
@@ -0,0 +1,67 @@
+
+## Technical Details
+
+### Server
+
+All server management will be written in Python. The server itself will be implemented using [bottle](http://bottlepy.org/docs/dev/index.html), a simple web framework with functionality to parse forms and JSON requests and responses. SQLite will be used to create and manage the database which holds the results.
+
+### Updates to Test Scripts
+
+The existing installation testing scripts, already written in Python, will be updated to send data to the server. Because the point of the scripts is to determine whether Python is configured correctly, the changes should not rely on any third party libraries and be compatible with both Python 2 and 3 so that even if a system fails a test, it will still be able to send the results. System data will be collected using the process library, which is included in Python's standard library. Data will be sent using the http.client and urllib.parse libraries.
+
+## Schedule of Deliverables
+
+### May 25th -  June 7th
+
+Create database schema and implement in SQLite.
+
+### June 8th - June 21th
+
+Write interface for database that allows high level manipulation of the database for future use within the server.
+
+### June 22nd - July 5th
+
+Begin work on server, with focus on handling POST requests, interpreting them, and correctly using database functions to transfer received information into SQLite database.
+
+### July 6th - July 19th
+
+Continue creating server, now focusing on aspects of how to retrieve information from database and serve it for analysis.
+
+### July 20th - August 2nd
+
+Modify the [first test script](https://github.com/wking/swc-setup-installation-test/blob/master/swc-installation-test-1.py) to submit simple pass/fail data about the Python version to the server.
+
+### August 3rd - August 16th
+
+Modify the [second test script](https://github.com/wking/swc-setup-installation-test/blob/master/swc-installation-test-2.py) to submit data to the server on the platform and the pass/fail status of dependencies and tools.
+
+### August 17th - August 21th 19:00 UTC
+
+Make a simple front end for easy viewing of data, as either a desktop or web application, which gets its information from the created API.
+
+## Future works
+
+If additional software dependencies are added to the installation scripts, then the server and database would have to be adapted to include them.
+
+## Open Source Development Experience
+
+This project with Software Carpentry will be my first experience with open source development.
```

Comment:
Unfortunately no, this will literally be my first time in open source software. The closest thing I can think of are some [solutions to programming challenges](https://github.com/rstolliker/challenges) and [an iOS app](https://github.com/rstolliker/FirstApp) that I have uploaded to GitHub, but they’re both fairly small side projects that I have done by myself. Would you recommend putting these under this section?